core/services/ocr3/plugins/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -0,0 +1,170 @@
+package ccipreader
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/crconsts"
+	ccipreaderpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/headtracker"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/ccip_reader_tester"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils/pgtest"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+func TestCCIPReader_CommitReportsGTETimestamp(t *testing.T) {}
+
+func TestCCIPReader_ExecutedMessageRanges(t *testing.T) {}
+
+func TestCCIPReader_MsgsBetweenSeqNums(t *testing.T) {}
+
+func TestCCIPReader_NextSeqNum(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	ctx := context.Background()
+
+	const chainS1 = cciptypes.ChainSelector(1)
+	const chainS2 = cciptypes.ChainSelector(2)
+	const chainS3 = cciptypes.ChainSelector(3)
+	const chainD = cciptypes.ChainSelector(4)
+
+	onChainSeqNums := map[cciptypes.ChainSelector]cciptypes.SeqNum{
+		chainS1: 10,
+		chainS2: 20,
+		chainS3: 30,
+	}
+
+	s := testSetup(t, ctx, chainD, onChainSeqNums)
+
+	cfg := evmtypes.ChainReaderConfig{
+		Contracts: map[string]evmtypes.ChainContractReader{
+			crconsts.ContractNameOffRamp: {
+				ContractABI: ccip_reader_tester.CCIPReaderTesterABI,
+				Configs: map[string]*evmtypes.ChainReaderDefinition{
+					crconsts.FunctionNameGetSourceChainConfig: {
+						ChainSpecificName: "getSourceChainConfig",
+						ReadType:          evmtypes.Method,
+					},
+				},
+			},
+		},
+	}
+
+	cr, err := evm.NewChainReaderService(ctx, lggr, s.lp, s.cl, cfg)
+	assert.NoError(t, err)
+	err = cr.Bind(ctx, []types.BoundContract{
+		{
+			Address: s.contractAddr.String(),
+			Name:    crconsts.ContractNameOffRamp,
+			Pending: false,
+		},
+	})
+	assert.NoError(t, err)
+	err = cr.Start(ctx)
```

Comment:
One liner for this is good. Also using require makes it stop instead of continuing with the rest of the tests afterwards and failing all of them making the logs pretty noisy.
```suggestion
	require.NoError(t, cr.Start(ctx))
```

Reply:
Same applies for the rest of the test